chain/client/src/client.rs

```diff
@@ -2211,11 +2212,7 @@ impl Client {
         let gas_price = cur_block_header.next_gas_price();
         let epoch_id = self.epoch_manager.get_epoch_id_from_prev_block(&head.last_block_hash)?;
         let shard_layout = self.runtime_adapter.get_shard_layout(&epoch_id)?;
-        let receiver_shard = account_id_to_shard_id(
-            self.epoch_manager.as_ref(),
-            tx.transaction.receiver_id(),
-            &epoch_id,
-        )?;
+        let receiver_shard = shard_layout.account_id_to_shard_id(tx.transaction.receiver_id());
```

Comment:
I would suggest getting the shard layout from epoch manager in the line above. The reason nayduck tests are failing is because the kv_runtime is used and it has some other shard layout hardcoded [here](https://github.com/near/nearcore/blob/c1f7fd20a92dc32941c61651b5e30ba6328e2a1d/chain/chain/src/test_utils/kv_runtime.rs#L509-L512). It's a bit crazy that those tests work as is. I don't think there is much value refactoring it at this point.